bridge/src/graphs/peg_out.rs

```
@@ -2408,15 +2387,15 @@ impl PegOutGraph {
         );
 
         // connector c pks = connector e1 pks + connector e2 pks
+        let commitment_public_keys = &merge_to_connector_c_commits_public_key(
+            connector_e1_commitment_public_keys,
+            connector_e2_commitment_public_keys,
+        );
         let connector_c = ConnectorC::new(
             network,
             operator_taproot_public_key,
-            &merge_to_connector_c_commits_public_key(
-                connector_e1_commitment_public_keys,
-                connector_e2_commitment_public_keys,
-            ),
-            lock_scripts_generator,
-            lock_scripts_copy,
+            commitment_public_keys,
+            ConnectorC::cache_id(commitment_public_keys).ok(),
```

I guess it's fine to ignore the error here and let the constructor regenerate the scripts. However, this leaves no trace of the error. Since there are no logs, we should probably print it to the console as an FYI to assist with potential debugging or diagnosis.

---

agreed

---

I think the original API was fine. Pls review https://github.com/elementlabs42/BitVM/pull/43/commits/753a803c30bd8baab8f6007e73f1421978f10e41

---

Also refined in https://github.com/elementlabs42/BitVM/pull/43/commits/3132a9adf01d6f3178a76c714ac4d77a1ff9c62c